src/LightPanels/LightPanels.js

```diff
@@ -530,17 +485,16 @@ module.exports = kind(
 	*/
 	replaceAt: function (start, count, info) {
 		var panels = this.getPanels(),
-			insertBefore, commonInfo, end, idx;
+			queue = this.popQueue,
+			insertBefore, commonInfo, end;
 
 		start = start < 0 ? panels.length + start : start;
 		end = start + count;
 		insertBefore = panels[end];
 		commonInfo = {addBefore: insertBefore};
+		queue = queue || [];
 
-		// remove existing panels
-		for (idx = end - 1; idx >= start; idx--) {
-			this.removePanel(panels[idx]);
-		}
+		queue = queue.concat(panels.splice(start, end - start));
```

Comment:
The intention was that we would modify `this.popQueue`, which we have a reference to if it exists, but this won't work for the case where we're creating a new array, so I'll fix this up. `panel.destroy()` will be called when we actually process our queue of panels to pop, if we are not caching views.